File: contador-de-palabras/contador.dart
void main() {
  String texto = "hola mundo hola dart mundo mundo";

  // Divide el texto en palabras
  List<String> listaPalabras = texto.split(' '); //recibe lo que quieres eliminar, en este caso los espacios vacios y retorna una lista
  print(listaPalabras);

  //Crea un mapa para contar la frecuencia de cada palabra
  Map frecuencia = {}; // creo un mapa vacio

  for (String palabra in listaPalabras) { //itero sobre la lista
    if (frecuencia.containsKey(palabra)) { //esto es similar un un palabra inLista
      frecuencia[palabra] += 1; // le sumo uno si la palabra ya esta en el mapa
    } else {
      frecuencia[palabra] = 1; //si no esta en el mapa lo igualo a uno
    }
  }
  print(frecuencia);

  //Encuentra la palabra más frecuente
  String palabraMayor = '';
  int conteo = 0;

  for (var entrada in frecuencia.entries) {
    if (entrada.value > conteo) {
      conteo = entrada.value;
      palabraMayor = entrada.key;
    }
  }
  //Imprime palabras que aparecen más de 1 vez
  print("La palabra mas frecuente es: ${palabraMayor} y aparece $conteo veces");
}

File: contador-de-palabras/contador.js
const texto = "hola mundo hola javascript mundo mundo";

// TODO:  Divide el texto en palabras usando split()
const listaPalabras = texto.split(' ');
// TODO: Crea un objeto para contar la frecuencia de cada palabra
const frecuencia = {};



for (let palabra of listaPalabras){
    if (palabra in frecuencia){
        frecuencia[palabra] += 1;
    }
    else {
        frecuencia[palabra] = 1;
    }
}
console.log(frecuencia);

// TODO: Encuentra la palabra más frecuente
let palabraFrecuente = "";
let contador = 0;

for (const [c, v] of Object.entries(frecuencia)){
    if( v > contador){
        contador = v;
        palabraFrecuente = c;
    }
}

// TODO: Filtra y muestra palabras que aparecen más de 1 vez
console.log(`La palabra mas frecuente es ${palabraFrecuente} y aparece ${contador}`);
